packages/smart-account/src/signers/SmartAccountSigner.ts

```diff
@@ -53,25 +40,28 @@ export class SmartAccountSigner extends Signer implements TypedDataSigner {
     return ethers.utils.getAddress(this._address)
   }
 
-  async signTransaction(signTransactionDto: SignTransactionDto): Promise<string> {
-    console.log(signTransactionDto)
-    const signature = ''
-    return signature
+  async getChainId(): Promise<number> {
+    return (await this.provider.getNetwork()).chainId
   }
 
-  // getProvider
-
-  getRelayer(chainId?: number): Promise<Relayer | undefined> {
-    console.log(chainId)
-    throw new Error('TODO')
+  async signTransaction(transaction: Deferrable<TransactionRequest>): Promise<string> {
+    if (!this.provider) {
+      throw new Error('missing provider')
+    }
+    const signature: any = await this.provider.send('eth_signTransaction', [transaction])
+    return signature
   }
 
-  // Review
-  // getProvider returns a Web3Provider instance for the current chain.
-  // Note that this method is bound to a particular chain
+  // getRelayer(chainId?: number): Promise<Relayer | undefined> {
+  //   console.log(chainId)
+  //   throw new Error('TODO')
+  // }
 
+  // Review getProvider
+  // getProvider returns a JsonRpcProvider instance for the current chain.
+  // Note that this method is bound to a particular chain
   // Review for the provider we want here
-  async getProvider(chainId?: number): Promise<Web3Provider | undefined> {
+  async getProvider(chainId?: number): Promise<JsonRpcProvider | undefined> {
```

Comment:
for example are this methods accessed anyway in any other classes where this will be passed as original signer?